### PR TITLE
Update PyPI links and capitalization in README.rst

### PR DESCRIPTION
**Pull request by mdickinson**
This PR:
- updates incorrect and inconsistent capitalization in the README
- adds a PyPI link for Sphinx
- adds a PyPI link for the Enthought Sphinx Theme (the previous link was to GitHub)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -54,13 +54,12 @@ Traits has the following optional dependencies:
 
 * `NumPy <http://pypi.python.org/pypi/numpy>`_ to support the trait types
   for arrays.
-* `Traitsui <https://pypi.python.org/pypi/traitsui>`_ to support Gui
+* `TraitsUI <https://pypi.python.org/pypi/traitsui>`_ to support Gui
```
Comment (rahulporuri) [inline README.rst:57]: nit : GUI instead of Gui? As long as this PR is about capitalization in README?

Comment (mdickinson) [reply README.rst:57]: Sure, let's do it!

Comment (mdickinson) [inline README.rst:57]:
```suggestion
* `TraitsUI <https://pypi.python.org/pypi/traitsui>`_ to support GUI
```

Comment (mdickinson) [reply README.rst]: Addressed.

**Comment by mdickinson** [comment]
I'll add this to the backport targets; it's not strictly a bugfix, but the resulting ugliness in the PyPI page has always bothered me.

**Comment by rahulporuri** [comment]
still LGTM.

**Pull request merged**